### PR TITLE
Add another bookmark view mode

### PR DESCRIPTION
**Pull request by slprime**
### 2 Bookmarks View Mode

| Default | New View |
|--------------|---------------|
| default | todo|

#### Toggle View Mode
view-mode-tooltip

view_mode

### CODE REVIEW

File: src/main/java/codechicken/nei/BookmarkPanel.java [hunk 1]
```diff
@@ -76,8 +76,147 @@ public BookmarkStackMeta(BookmarkRecipeId recipeId, int count, boolean ingredien
 
     protected static class BookmarkGrid extends ItemsGrid
     {
+        public final static int VIEW_MODE_BOOKMARKS = 0;
```
Comment (eigenraven) [inline src/main/java/codechicken/nei/BookmarkPanel.java:79]: This should be an enum

File: src/main/java/codechicken/nei/BookmarkPanel.java [hunk 2]
```diff
@@ -76,8 +76,147 @@ public BookmarkStackMeta(BookmarkRecipeId recipeId, int count, boolean ingredien
...
+        protected List<Integer> getMask()
+        {
+            if (gridMask != null) {
+                return gridMask.size() > page? gridMask.get(page): new ArrayList<>();
+            }
+
+            return super.getMask();
+        }
+
+        private void sortingIngredients(final boolean forward)
```
Comment (eigenraven) [inline src/main/java/codechicken/nei/BookmarkPanel.java:131]:
```suggestion
        private void sortIngredients(final boolean forward)
```

File: src/main/java/codechicken/nei/BookmarkPanel.java [hunk 3]
```diff
@@ -91,6 +230,19 @@ public int indexOf(ItemStack stackA, BookmarkRecipeId recipeId)
     
             return -1;
         }
+            
+        protected int getSlotIndex(int idx)
+        {
+            List<Integer> mask = getMask();
+
+            for (int i = 0; i < mask.size(); i++) {
+                if (mask.get(i) != null && mask.get(i) == idx) {
```
Comment (eigenraven) [inline src/main/java/codechicken/nei/BookmarkPanel.java:239]:
```suggestion
                if (mask.get(i) != null && mask.get(i).intValue() == idx) {
```
Boxed integers might be different references even if they have the same value

**Review by eigenraven — CHANGES_REQUESTED** [review]
Minor changes

File: src/main/java/codechicken/nei/BookmarkPanel.java (hunk 2 above)
Comment (slprime) [reply src/main/java/codechicken/nei/BookmarkPanel.java]: change

**Review by eigenraven — APPROVED** [review]

**Pull request merged**